libwild/src/version_script.rs

```diff
@@ -552,12 +579,74 @@ fn parse_token<'input>(input: &mut &'input BStr) -> winnow::Result<&'input [u8]>
     take_while(1.., |b| !b" (){}\n\t".contains(&b)).parse_next(input)
 }
 
+enum GlobPatternType {
+    Exact,
+    Star,
+    NonStar,
+}
+
+fn analyze_glob_pattern(pattern: &[u8]) -> GlobPatternType {
+    let mut pattern_type = GlobPatternType::Exact;
+    let mut it = pattern.iter();
+
+    while let Some(&c) = it.next() {
+        match c {
+            b'\\' => {
+                it.next();
```

Comment:
If we get here and don't subsequently encounter a `*`, we know that the pattern has an escape sequence. Consider adding another variant to `GlobPatternType` - e.g. `EscapedExact`. Then you wouldn't need the `has_escape_chars` function.